enhancements/compound-bundle-constraints.md

```diff
@@ -0,0 +1,350 @@
+---
+title: compound-bundle-constraints
+authors:
+  - "@estroz"
+reviewers:
+  - "@joelanford"
+  - "@kevinrizza"
+  - "@dinhxuanvu"
+approvers:
+  - "@joelanford"
+  - "@kevinrizza"
+  - "@dinhxuanvu"
+creation-date: 2021-10-14
+last-updated: 2021-10-14
+status: implementable
+see-also:
+  - "/enhancements/operator-dependency-resolution.md"
+replaces:
+superseded-by:
+---
+
+# compound-bundle-constraints
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Open Questions
+
+1. Eventually I think it would be a good idea to create a general `constraints` property
+because, as we add more constraint properties to the top-level bundle config, it becomes
+difficult to differentiate what is a property (an intrinsic property of a bundle) and
+what is a property that acts as a constraint (a property of a bundle _describing other packages_).
+Is this actually a good idea, and is this outside the scope of this EP?
+  - From what I have heard, this is outside the scope of this EP. Would like input though.
```

Comment:
IMO, there needs to be _some_ mechanism for OLM to distinguish arbitrary properties from constraints, because it is inevitable that we'll get feature requests that dictate that the resolver doesn't simply ignore them. I think this is a fundamental prerequisite for both this EP and for the generic constraint EP in #91.